layer.go

```diff
@@ -146,6 +146,31 @@ type Layer struct {
 	Exec Exec `toml:"-"`
 }
 
+func (l Layer) Reset() (Layer, error) {
+	l.LayerTypes = LayerTypes{
+		Build:  false,
+		Launch: false,
+		Cache:  false,
+	}
+
+	l.SharedEnvironment = Environment{}
+	l.BuildEnvironment = Environment{}
+	l.LaunchEnvironment = Environment{}
+	l.Metadata = nil
+
```

Comment:
Do we also want to reset Profile?

Reply:
Added both Profile and Exec to the reset logic. Exec might not be necessary but I figured I would add it for consistency sake.